### PR TITLE
fix: allow setting flags for notify action

### PR DESCRIPTION
**Pull request by jamil314**
Notify action config is copied from declare action config
Demoland PR: https://github.com/opencrvs/opencrvs-demoland/pull/18

## Description

Clearly describe what has been changed. Include relevant context or background.
Explain how the issue was fixed (if applicable) and the root cause.

Link this pull request to the GitHub issue (and optionally name the branch `ocrvs-<issue #>`)

## Checklist

- [ ] I have linked the correct Github issue under "Development"
- [ ] I have tested the changes locally, and written appropriate tests
- [ ] I have tested beyond the happy path (e.g. edge cases, failure paths)
- [ ] I have updated the changelog with this change (if applicable)
- [ ] I have updated the GitHub issue status accordingly

### CODE REVIEW

File: packages/commons/src/events/state/flags.ts [hunk 1]
```diff
@@ -121,7 +121,7 @@ export function resolveEventCustomFlags(
 
   return actions.reduce((acc, action, idx) => {
     let actionConfig
-    if (isActionConfigType(action.type)) {
+    if (isActionConfigType(action.type) || action.type === ActionType.NOTIFY) {
```
Comment (cibelius) [inline packages/commons/src/events/state/flags.ts:124]: I'm afraid that this might have some unintended consequences, for example:

If now a Registration Officer or Registrar would Notify a record (though I'm not sure if they can), the `validated` flag would be applied, because we have:

```
    {
      type: ActionType.DECLARE,
      // ...
      flags: [
        {
          id: 'validated',
          operation: 'add',
          conditional: or(
            user.hasRole('REGISTRATION_AGENT'),
            user.hasRole('LOCAL_REGISTRAR')
          )
        },
```

But I think we want to apply the 'validated' flag only if the record is declared, not when its notified.

Comment (cibelius) [reply packages/commons/src/events/state/flags.ts:124]: To prevent this, what we can do is add separate configuration for ActionType.NOTIFY (instead of it using ActionType.DECLARE config). This is quite a significant change though, and it would need to be handled in codemod as well.

Comment (jamil314) [reply packages/commons/src/events/state/flags.ts:124]: I'll see if I can push back on adding the new flag on notify action. Otherwise i'll adopt your suggestion.

Comment (rikukissa) [reply packages/commons/src/events/state/flags.ts:124]: Good point @cibelius. Let's keep all actions as separate as possible and never implicitly share config

Comment (cibelius) [reply packages/commons/src/events/state/flags.ts:124]: @rikukissa we created an issue for this here: https://github.com/opencrvs/opencrvs-core/issues/12707

**Comment by Zangetsu101** [comment]
@jamil314 Can we close this one?
